tests/CMakeLists.txt

```diff
@@ -28,7 +28,7 @@ endif()
 
 add_compile_options(${CXX_WARNING_ARGS})
 
-add_test(cppcheck cppcheck --enable=all -USCHEMA_DIR --error-exitcode=2 --inline-suppr -I${CMAKE_SOURCE_DIR} ${CMAKE_SOURCE_DIR}/src ${CMAKE_SOURCE_DIR}/tests)
+add_test(cppcheck cppcheck --force --enable=all -USCHEMA_DIR --error-exitcode=2 --suppress=missingInclude --suppress=unusedFunction --suppress=unreadVariable --library=qt --inline-suppr -I${CMAKE_SOURCE_DIR} ${CMAKE_SOURCE_DIR}/src ${CMAKE_SOURCE_DIR}/tests)
```

Comment:
I would like to avoid to suppress these, if any of them trigger that should be a something we fix not *ignore* imo

Reply:
@mariogrip 

Can I push the whole thing?

Reply:
@tari01: please don't, yet. We should get the cppcheck fixed entirely first.

@mariogrip: I did a cppcheck --check-config and I get this:

```
ome/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/adbd-client.cpp:26]: (information) Include file: <atomic> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/adbd-client.cpp:27]: (information) Include file: <cctype> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/adbd-client.cpp:28]: (information) Include file: <cstring> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/adbd-client.cpp:29]: (information) Include file: <chrono> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/adbd-client.cpp:30]: (information) Include file: <condition_variable> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/adbd-client.cpp:31]: (information) Include file: <mutex> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/adbd-client.cpp:32]: (information) Include file: <thread> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: 1/25 files checked 8% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/exporter.cpp ...
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/indicator.h:22]: (information) Include file: <core/property.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/indicator.h:24]: (information) Include file: <gio/gio.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/indicator.h:26]: (information) Include file: <string> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/indicator.h:27]: (information) Include file: <vector> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/exporter.h:25]: (information) Include file: <core/signal.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/exporter.h:27]: (information) Include file: <memory> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: 2/25 files checked 13% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/greeter.cpp ...
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/greeter.h:22]: (information) Include file: <core/property.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/greeter.h:24]: (information) Include file: <memory> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/greeter.h:25]: (information) Include file: <string> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/greeter.cpp:23]: (information) Include file: <gio/gio.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: 3/25 files checked 18% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/indicator.cpp ...
1: 4/25 files checked 19% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/main.cpp ...
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/rotation-lock.h:25]: (information) Include file: <memory> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-monitor.h:22]: (information) Include file: <core/signal.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-monitor.h:24]: (information) Include file: <memory> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-monitor.h:25]: (information) Include file: <string> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-manager.h:25]: (information) Include file: <memory> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-manager.h:26]: (information) Include file: <string> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/main.cpp:27]: (information) Include file: <glib/gi18n.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/main.cpp:28]: (information) Include file: <gio/gio.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/main.cpp:30]: (information) Include file: <locale.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: 5/25 files checked 21% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/rotation-lock.cpp ...
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/rotation-lock.cpp:22]: (information) Include file: <glib/gi18n.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: 6/25 files checked 25% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-manager.cpp ...
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-snap.h:24]: (information) Include file: <core/signal.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-snap.h:26]: (information) Include file: <memory> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-snap.h:27]: (information) Include file: <string> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-manager.cpp:24]: (information) Include file: <glib.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-manager.cpp:26]: (information) Include file: <sys/types.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-manager.cpp:27]: (information) Include file: <sys/stat.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-manager.cpp:28]: (information) Include file: <fcntl.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-manager.cpp:29]: (information) Include file: <unistd.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-manager.cpp:31]: (information) Include file: <set> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: 7/25 files checked 30% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-monitor.cpp ...
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-monitor.cpp:22]: (information) Include file: <glib.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-monitor.cpp:23]: (information) Include file: <gudev/gudev.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: 8/25 files checked 32% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-snap.cpp ...
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-snap.cpp:23]: (information) Include file: <glib/gi18n.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/src/usb-snap.cpp:24]: (information) Include file: <gio/gio.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: 9/25 files checked 39% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/gmock/CMakeFiles/3.13.4/CompilerIdC/CMakeCCompilerId.c ...
1: 10/25 files checked 55% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/gmock/CMakeFiles/3.13.4/CompilerIdCXX/CMakeCXXCompilerId.cpp ...
1: 11/25 files checked 70% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/gmock/CMakeFiles/feature_tests.c ...
1: 12/25 files checked 70% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/gmock/CMakeFiles/feature_tests.cxx ...
1: 13/25 files checked 78% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/integration/usb-manager-test.cpp ...
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/adbd-server.h:20]: (information) Include file: <gio/gio.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/adbd-server.h:21]: (information) Include file: <gio/gunixsocketaddress.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/adbd-server.h:23]: (information) Include file: <string> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/adbd-server.h:24]: (information) Include file: <thread> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/adbd-server.h:25]: (information) Include file: <vector> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/dbus-types.h:21]: (information) Include file: <QDBusMetaType> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/dbus-types.h:22]: (information) Include file: <QtCore> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/dbus-types.h:23]: (information) Include file: <QString> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/dbus-types.h:24]: (information) Include file: <QVariantMap> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/glib-fixture.h:22]: (information) Include file: <functional> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/glib-fixture.h:23]: (information) Include file: <map> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/glib-fixture.h:24]: (information) Include file: <memory> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/glib-fixture.h:26]: (information) Include file: <glib.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/glib-fixture.h:27]: (information) Include file: <glib/gstdio.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/glib-fixture.h:28]: (information) Include file: <gio/gio.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/glib-fixture.h:30]: (information) Include file: <gtest/gtest.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/glib-fixture.h:32]: (information) Include file: <locale.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/gtest-qt-print-helpers.h:4]: (information) Include file: <QDBusObjectPath> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/gtest-qt-print-helpers.h:5]: (information) Include file: <QString> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/gtest-qt-print-helpers.h:6]: (information) Include file: <QStringList> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/gtest-qt-print-helpers.h:7]: (information) Include file: <QVariant> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/qt-fixture.h:28]: (information) Include file: <gtest/gtest.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/qt-fixture.h:30]: (information) Include file: <QDBusArgument> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/qt-fixture.h:31]: (information) Include file: <QVariant> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/qt-fixture.h:32]: (information) Include file: <QSignalSpy> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/integration/usb-manager-test.cpp:28]: (information) Include file: <libqtdbustest/DBusTestRunner.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/integration/usb-manager-test.cpp:29]: (information) Include file: <libqtdbustest/QProcessDBusService.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/integration/usb-manager-test.cpp:30]: (information) Include file: <libqtdbusmock/DBusMock.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/integration/usb-manager-test.cpp:32]: (information) Include file: <fstream> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/integration/usb-manager-test.cpp:33]: (information) Include file: <sstream> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/integration/usb-manager-test.cpp:34]: (information) Include file: <vector> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: 14/25 files checked 86% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/integration/usb-manager-test_autogen/mocs_compilation.cpp ...
1: 15/25 files checked 86% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/unit/adbd-client-test.cpp ...
1: 16/25 files checked 89% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/unit/adbd-client-test_autogen/mocs_compilation.cpp ...
1: 17/25 files checked 89% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/unit/greeter-test.cpp ...
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/unit/greeter-test.cpp:26]: (information) Include file: <libqtdbustest/DBusTestRunner.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/unit/greeter-test.cpp:27]: (information) Include file: <libqtdbustest/QProcessDBusService.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/unit/greeter-test.cpp:28]: (information) Include file: <libqtdbusmock/DBusMock.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: 18/25 files checked 93% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/unit/greeter-test_autogen/mocs_compilation.cpp ...
1: 19/25 files checked 93% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/unit/rotation-lock-test.cpp ...
1: 20/25 files checked 94% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/unit/rotation-lock-test_autogen/mocs_compilation.cpp ...
1: 21/25 files checked 94% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/unit/usb-snap-test.cpp ...
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/unit/usb-snap-test.cpp:25]: (information) Include file: <libqtdbustest/DBusTestRunner.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/unit/usb-snap-test.cpp:26]: (information) Include file: <libqtdbustest/QProcessDBusService.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/unit/usb-snap-test.cpp:27]: (information) Include file: <libqtdbusmock/DBusMock.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: 22/25 files checked 98% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/unit/usb-snap-test_autogen/mocs_compilation.cpp ...
1: 23/25 files checked 98% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/qmain.cpp ...
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/qmain.cpp:22]: (information) Include file: <QCoreApplication> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/qmain.cpp:23]: (information) Include file: <QTimer> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/qmain.cpp:24]: (information) Include file: <gtest/gtest.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/qmain.cpp:26]: (information) Include file: <libqtdbusmock/DBusMock.h> not found. Please note: Cppcheck does not need standard library headers to get proper results.
1: [/home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/qmain.cpp:60]: (information) Include file: "qmain.moc" not found.
1: 24/25 files checked 99% done
1: Checking /home/mike/MyDocuments/4projects/ayatana-upstream/ayatana-indicator-display.upstream/ayatana-indicator-display/tests/utils/test-utils_autogen/mocs_compilation.cpp ...
1: 25/25 files checked 100% done
1/1 Test #1: cppcheck .........................   Passed    0.46 sec

The following tests passed:
	cppcheck

100% tests passed, 0 tests failed out of 1

Total Test time (real) =   0.47 sec
```

I am not really experienced with cppcheck, but from what I see, I guess, we need to add all the various include paths of the single code files (e.g. glib2.0's includepath, etc.) need. As the sterr output says, we don't need the system's normal include paths added, only the special share libs (e.g. glib2.0, gio, etc.).

Reply:
@mariogrip: are there any other cmake projects on the ubports name space where include file location is working with cppcheck?